regexp.go

```diff
@@ -371,3 +371,29 @@ func (re *Regexp) GroupNumberFromName(name string) int {
 
 	return -1
 }
+
+// MarshalText implements [encoding.TextMarshaler]. The output
+// matches that of calling the [Regexp.String] method.
+func (re *Regexp) MarshalText() ([]byte, error) {
+	return []byte(re.String()), nil
+}
+
+// UnmarshalText implements [encoding.TextUnmarshaler] by calling
+// [Compile] on the encoded value.
+func (re *Regexp) UnmarshalText(text []byte) error {
+	newRE, err := Compile(string(text), re.options)
```

Comment:
I think the options should be 0 here since there's no real valid way to set `re.options` anyway.

Reply:
I'm also thinking about this. Is it possible to provide a global flag, such as DefaultUnmarshalOptions?

Reply:
That could work.